File: src/Web/WebApi/src/service/InterestService.js
export default class InterestService {
    constructor(request) {
        this.request = request;
    }

    async getList() {
        const response = await this.request({
            method: 'get',
            url: '/interest/list'
        });
        return response?.data ?? response;
    }

    async savePartyInterests(email, interestIds) {
        const response = await this.request({
            method: 'post',
            url: '/partyInterest/save',
            data: { email, interestIds }
        });
        return response?.data ?? response;
    }

    async getMyInterests(email) {
        const response = await this.request({
            method: 'get',
            url: '/partyInterest/my-interests',
            ...(email ? { params: { email } } : {})
        });
        return response?.data ?? response;
    }
}

File: src/Mobile/lib/services/interest_service.dart
import 'package:dio/dio.dart';

class InterestService {
  InterestService(this._dio);

  final Dio _dio;

  static Options _okOnly() => Options(
        validateStatus: (s) => s != null && s >= 200 && s < 300,
      );

  Future<dynamic> getList() async {
    final r = await _dio.get<dynamic>('/interest/list');
    return r.data;
  }

  Future<dynamic> savePartyInterests({
    required String email,
    required List<int> interestIds,
  }) async {
    final r = await _dio.post<dynamic>(
      '/partyInterest/save',
      data: {'email': email, 'interestIds': interestIds},
    );
    return r.data;
  }

  /// [email] isteğe bağlı; Authorization Bearer ile gönderildiğinde sunucu JWT içindeki e-postayı kullanır.
  Future<dynamic> getMyInterests({String? email}) async {
    final r = await _dio.get<dynamic>(
      '/partyInterest/my-interests',
      queryParameters: {
        if (email != null && email.trim().isNotEmpty) 'email': email.trim(),
      },
      options: _okOnly(),
    );
    return r.data;
  }
}
